analysis.go

```diff
@@ -50,60 +49,51 @@ type DeploymentConfigs struct {
 	} `json:"items"`
 }
 
-type CheckTask func(string, io.Writer) (Result, error)
+type CheckTask func(string, string) (Result, error)
+
+func GetAnalysisTasks(tasks chan<- Task, basepath string, projects []string, results chan<- CheckResults) {
```

Comment:
I understand we need a results channel, but why does the caller need to create the channel and pass it in as an argument?

Granted I might be missing something, I thought of `GetAnalysisTasks` contract as "I'll send tasks to a channel, and here's where you can look for the results of those tasks if you want".

Generally speaking, I felt there is some difficulty to arrange for closing the results channel... how to tell that no more results will be sent. I might need to reason more about https://github.com/feedhenry/fh-system-dump-tool/pull/36/files/9ef911e9525d4008c952979fbf3bfd5514455aa0#diff-2628f4a06e8be3343e85abecc49fae35R193